src/clj/puppetlabs/puppetserver/certificate_authority.clj

```diff
@@ -795,15 +795,14 @@
       {:kind :invalid-subject-name
        :msg  (i18n/tru "Subject contains a wildcard, which is not allowed: {0}" subject)}))
 
-  (when (or (str/ends-with? subject ".")
-            (str/ends-with? subject "-"))
+  (when (str/ends-with? subject "-")
```

Comment:
IS this important for logging? I think the check on line 811 would catch this https://github.com/puppetlabs/puppetserver/pull/2700/files#diff-ac2e29a707d8bb0a3cca1bdb78e91e56a4a9a9213fb1fde7839b628837f0cab1R811

Reply:
I think from a support standpoint it is useful output.  The other message is more generic and doesn't describe WHY it is invalid.  I don't feel strongly about it, and I do think we need to have much better logging diagnostics in our applications in general.